extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -153,6 +153,15 @@ export class AmpAdApiHandler {
           {requestedHeight: height, requestedWidth: width},
           targetOrigin,
           this.is3p);
+    }).catch(() => {
+      const targetOrigin =
+          this.iframe_.src ? parseUrl(this.iframe_.src).origin : '*';
+      postMessage(
+          this.iframe_,
+          'embed-size-denied',
+          {requestedHeight: height, requestedWidth: width},
+          targetOrigin,
+          this.is3p_);
```

Comment:
There's a lot of duplication here, maybe make a helper function that takes the `iframe` and message type?